presto-hive/src/main/java/com/facebook/presto/hive/HivePageSourceProvider.java

```diff
@@ -447,6 +481,67 @@ public static Optional<ConnectorPageSource> createHivePageSource(
             }
         }
 
+        return getPageSourceFromCursorProvider(
```

Comment:
@codope so I was just trying to understand the reason behind calling this method again here. This is already called on line 414 after checking some preconditions. Is it really needed to call it here? Your inputs are appreciated. :)

Reply:
`getPageSourceFromCursorProvider` just extracts out the common processing logic and is called again here for completeness (no missing return statement). The call on line 414 is there so as to ignore page source factories if a custom record reader is to be used.